File: DotNetGotchas/VBDN/RTTI/Modified3/RTTIExample/Cat.vb
'Cat.vb
Public Class Cat
    Inherits Animal

    Public Sub Meow()
        Console.WriteLine("Cat Meowing")
    End Sub

    Public Overrides Sub MakeNoise()
        Meow()
    End Sub

    Public Overrides Function Play(ByVal other As Animal) As Boolean
        Return False
    End Function
End Class

File: DotNetGotchas/CSharp/RTTI/Modified3/RTTIExample/Cat.cs
//Cat.cs
using System;

namespace RTTIExample
{
	public class Cat : Animal
	{
		public void Meow()
		{
			Console.WriteLine("Cat Meowing");
		}

		public override void MakeNoise()
		{
			Meow();
		}

		public override bool Play(Animal other)
		{
			return false;
		}

	}
}
